test/test_datapoints.py

```diff
@@ -186,146 +187,115 @@ def test_other_op_no_wrapping(return_type):
-        datapoints.Mask(torch.randint(0, 256, (16, 16), dtype=torch.uint8)),
-    ],
-)
+@pytest.mark.parametrize("make_input", [make_image, make_bounding_box, make_segmentation_mask, make_video])
 @pytest.mark.parametrize("requires_grad", [False, True])
-def test_deepcopy(datapoint, requires_grad):
-    if requires_grad and not datapoint.dtype.is_floating_point:
-        return
+def test_deepcopy(make_input, requires_grad):
+    dp = make_input(dtype=torch.float)
 
-    datapoint.requires_grad_(requires_grad)
+    dp.requires_grad_(requires_grad)
 
-    datapoint_deepcopied = deepcopy(datapoint)
+    dp_deepcopied = deepcopy(dp)
 
-    assert datapoint_deepcopied is not datapoint
-    assert datapoint_deepcopied.data_ptr() != datapoint.data_ptr()
-    assert_equal(datapoint_deepcopied, datapoint)
+    assert dp_deepcopied is not dp
+    assert dp_deepcopied.data_ptr() != dp.data_ptr()
+    assert_equal(dp_deepcopied, dp)
 
-    assert type(datapoint_deepcopied) is type(datapoint)
-    assert datapoint_deepcopied.requires_grad is requires_grad
+    assert type(dp_deepcopied) is type(dp)
+    assert dp_deepcopied.requires_grad is requires_grad
 
 
+@pytest.mark.parametrize("make_input", [make_image, make_bounding_box, make_segmentation_mask, make_video])
 @pytest.mark.parametrize("return_type", ["Tensor", "datapoint"])
-def test_operations(return_type):
-    datapoints.set_return_type(return_type)
+@pytest.mark.parametrize(
+    "op",
+    (
+        lambda dp: dp + torch.rand(*dp.shape),
+        lambda dp: torch.rand(*dp.shape) + dp,
+        lambda dp: dp * torch.rand(*dp.shape),
+        lambda dp: torch.rand(*dp.shape) * dp,
+        lambda dp: dp + 3,
+        lambda dp: 3 + dp,
+        lambda dp: dp + dp,
+        lambda dp: dp.sum(),
+        lambda dp: dp.reshape(-1),
+        lambda dp: dp.int(),
+        lambda dp: torch.stack([dp, dp]),
+        lambda dp: torch.chunk(dp, 2)[0],
+        lambda dp: torch.unbind(dp)[0],
+    ),
+)
+def test_usual_operations(make_input, return_type, op):
+
+    dp = make_input()
+    with datapoints.set_return_type(return_type):
+        out = op(dp)
+    assert type(out) is (type(dp) if return_type == "datapoint" else torch.Tensor)
+    if isinstance(dp, datapoints.BoundingBoxes) and return_type == "datapoint":
+        assert hasattr(out, "format")
+        assert hasattr(out, "canvas_size")
+
+
+def test_subclasses():
+    img = make_image()
+    masks = make_segmentation_mask()
+
+    with pytest.raises(TypeError, match="unsupported operand"):
+        img + masks
+
+
+def test_set_return_type():
+    img = make_image()
+
+    assert type(img + 3) is torch.Tensor
+
+    with datapoints.set_return_type("datapoint"):
+        assert type(img + 3) is datapoints.Image
+    assert type(img + 3) is torch.Tensor
+
+    datapoints.set_return_type("datapoint")
+    assert type(img + 3) is datapoints.Image
```

Comment:
Should we switch off return type to tensor after the last assert (ln 301) to prevent it to leak in the other tests?

Reply:
There is an autouse fixture that makes sure we always go back to "tensor return type" on this file 


https://github.com/pytorch/vision/blob/885917174099af449fd26644a65f7fec43450476/test/test_datapoints.py#L19-L22

Reply:
Not blocking, but this way of doing things is far from being the most explicit and obvious

Reply:
I added manual cleanups in [89b9dff](https://github.com/pytorch/vision/pull/7828/commits/89b9dffe8640e8261968a088627a02888a7a7009)